### PR TITLE
CRM-19243: Free membership results in two email receipts

### PR DESCRIPTION
**Pull request by jitendrapurohit**
* [CRM-19243: Free membership results in two activities and email receipts](https://issues.civicrm.org/jira/browse/CRM-19243)

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -2561,6 +2561,11 @@ public function composeMessageArray(&$input, &$ids, &$values, $returnMessageText
             if ($membership_status == 'Pending' && $membership->is_pay_later == 1) {
               $values['is_pay_later'] = 1;
             }
+            // Pass amount to floatval as string '0.00' is considered a
+            // valid amount and includes Fee section in the mail.
+            if (isset($values['amount'])) {
+              $values['amount'] = floatval($values['amount']);
+            }
```
Comment (jitendrapurohit) [inline CRM/Contribute/BAO/Contribution.php]: **api_v3_ContributionPageTest**: 
`testSubmitMembershipBlockIsSeparatePaymentZeroDollarsPayLaterWithEmail` and 
`testSubmitMembershipBlockNotSeparatePaymentZeroDollarsWithEmail` asserts this change after modification to the second file included in this PR.

Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: Do either of the tests check how many emails are sent?

Comment (jitendrapurohit) [inline CRM/Contribute/BAO/Contribution.php]: The latest commit checks that too. Thanks!

**Review by seamuslee001 — APPROVED** [review]
This makes sense to me and happy that the test ensures only 1 email gets sent

**Pull request merged**